src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnEuropeConsentViewController.swift

```diff
@@ -0,0 +1,181 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import UIKit
+import Combine
+
+final class ExposureSubmissionWarnEuropeConsentViewController: DynamicTableViewController, ENANavigationControllerWithFooterChild {
+
+	typealias PrimaryButtonHandler = (Bool, @escaping (Bool) -> Void) -> Void
+
+	// MARK: - Init
+
+	init?(
+		coder: NSCoder,
+		onPrimaryButtonTap: @escaping PrimaryButtonHandler
+	) {
+		self.onPrimaryButtonTap = onPrimaryButtonTap
+
+		super.init(coder: coder)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - Overrides
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		setupView()
+	}
+
+	// MARK: - Protocol ENANavigationControllerWithFooterChild
+
+	func navigationController(_ navigationController: ENANavigationControllerWithFooter, didTapPrimaryButton button: UIButton) {
+		onPrimaryButtonTap(consentGiven) { [weak self] isLoading in
+		   self?.navigationFooterItem?.isPrimaryButtonLoading = isLoading
+		   self?.navigationFooterItem?.isPrimaryButtonEnabled = !isLoading
+		}
+	}
+
+	// MARK: - Private
+
+	private let onPrimaryButtonTap: PrimaryButtonHandler
+
+	@Published var consentGiven: Bool = false
+	private var consentSubscription: AnyCancellable?
+
+	private func setupView() {
+		navigationItem.title = AppStrings.ExposureSubmissionWarnEuropeConsent.title
+		navigationFooterItem?.primaryButtonTitle = AppStrings.ExposureSubmissionWarnEuropeConsent.continueButton
+
+		setupTableView()
+	}
+
+	private func setupTableView() {
+		tableView.delegate = self
+		tableView.dataSource = self
+
+		tableView.register(
+			DynamicTableViewRoundedCell.self,
+			forCellReuseIdentifier: CustomCellReuseIdentifiers.roundedCell.rawValue
+		)
+
+		dynamicTableViewModel = dynamicTableViewModel()
+	}
+
+	private func dynamicTableViewModel() -> DynamicTableViewModel {
+		DynamicTableViewModel.with {
```

Comment:
Would be nice to let this live in its own ViewModel file :)

Reply:
This is a change that is showing up here which i pulled over from develop, so I have to find out who was responsible for this :-D But somebody already approved it and it got into develop heehee

Reply:
I think I will add it to our refactoring epic 👍

Reply:
Ah okay, thanks :)